lib/uplink/buckets.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+	"errors"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// BucketOpts holds the cipher, path, key, and enc. scheme for each bucket since they
+// can be different for each
+type BucketOpts struct {
+	PathCipher       storj.Cipher
+	EncPathPrefix    storj.Path
```

Comment:
I don't think buckets have a top-level encrypted path prefix associated with them. I think when deriving an encrypted path you start with an hmac of the bucket's encryption key and the bucket's name.

Reply:
So I can just remove EncPathPrefix?

Reply:
they do if you're only granted access to a subfolder, but i agree bucketopts is not the struct for it.

Reply:
Checking..... yes, it looks like you can do that. You only need an encrypted path prefix if you're dealing with a "directory" (which is just a virtual thing as far as the object store is concerned, but would affect the encrypted path names of files created within it).